clients/src/main/java/org/apache/kafka/common/record/ControlRecordType.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.common.record;
+
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.protocol.types.Struct;
+import org.apache.kafka.common.protocol.types.Type;
+
+import java.nio.ByteBuffer;
+
+public enum ControlRecordType {
+    COMMIT((short) 0), ABORT((short) 1), UNKNOWN((short) -1);
+
+    private static final short CURRENT_CONTROL_RECORD_KEY_VERSION = 0;
+    private static final Schema CONTROL_RECORD_KEY_SCHEMA_VERSION0 = new Schema(
+            new Field("version", Type.INT16),
+            new Field("type", Type.INT16));
+
+    private final short type;
+
+    ControlRecordType(short type) {
+        this.type = type;
+    }
+
+    public Struct recordKey() {
+        if (this == UNKNOWN)
```

Comment:
Synced offline. The intended use of `UNKNOWN` is for backward compatibility: if a client doesn't know about some future new control message, then it will get UNKNOWN and should be able to handle that. This makes sense.